process/block/shardblock.go

```diff
@@ -1088,8 +1010,15 @@ func (sp *shardProcessor) getProcessedMetaBlocksFromMiniBlockHashes(
 			processedMetaHdrs = append(processedMetaHdrs, metaBlock)
 		}
 	}
+	sp.hdrsForCurrBlock.mutHdrsForBlock.RUnlock()
 
-	return processedMetaHdrs, processedMBs, nil
+	if len(processedMetaHdrs) > 1 {
+		sort.Slice(processedMetaHdrs, func(i, j int) bool {
```

Comment:
There are a lot of sort.Slice in this file, all sorting after the nonce. Could they be extracted into a method?

Reply:
Done